### ISSUE TITLE
IP Range DNS Lookup

### ISSUE THREAD

**Issue by dreinull** [mlocati/ip-lib#49]
Following the idea of #44, I would love to gave the possibility to get the reverse pointer for an IP range.

```php
Factory::rangeFromString('cafe::/32')->getReverseDNSLookupName();

// 0.0.0.0.e.f.a.c.in-addr.arpa
```

Is this maybe in the scope of this package? Would love to contribute to this idea.

**Comment by mlocati** [comment]
I've never seen the requirement to have the reverse lookup pointer of a range...

What would be the reverse lookup pointer of this?

```php
Factory::rangeFromString('cafe::/0')->getReverseDNSLookupName();
```

**Comment by dreinull** [comment]
Working for a hosting company, we need to create ReverseDNS zones in the RIPE database and on our name servers for the networks we delegated to our customers.

So let's say, we own the network `cafe::/16` and delegated the network `cafe:affe::/32` to our customer we need to create an entry on our name server like `e.f.f.a.e.f.a.c.in-addr.arpa` pointing to that customer.

To answer your question: This example does not really make sense to me as it is a non-practical case, but a tool I am using says the following:

```bash
$ ipv6calc -a cafe::/0
ip6.arpa.
```

**Comment by tobzsc** [comment]
I see the same issue for our ReverseDNS environment. We basically add IPv6 zones based on subnets. It would be nice if there would be support to get a nibble representation from a subnet:

`$ ipv6calc -a 2001:f89::/32
9.8.f.0.1.0.0.2.ip6.arpa.`

In this example we are getting the correct nibble zone representation from a subnet. Would be nice to see this feature!

**Comment by mlocati** [comment]
Is there an RFC describing reverse pointers for IPv4 and IPv6?

**Comment by dreinull** [comment]
Couldn't find any. A good description can be found here: https://www.ripe.net/manage-ips-and-asns/db/support/configuring-reverse-dns

**Comment by mlocati** [comment]
What about #53?

See [the new tests](https://github.com/mlocati/ip-lib/blob/6f8bfb808175cb6c68572e7658d7cfaa5161da40/test/tests/Ranges/ReverseDNSLookupNameTest.php#L13-L187) to see the results of this new method.

**Comment by dreinull** [comment]
This looks good for me. I am really looking forward to use this. :)

**Comment by mlocati** [comment]
This new `getReverseDNSLookupName` method is included in the brand new 1.13.0 version
